### ISSUE TITLE
"Removing non-existent queue" message in logs when attempting to publish to SNS

### ISSUE THREAD

**Issue by sineqa** [localstack/localstack#2881]
# This is a 
[x] bug report
[ ] feature request

Version 0.11.4

# Detailed description
Trying to publish to SNS and I keep getting "Removing non-existent queue" in the logs - and when I publish to SNS the messages don't end up on the queue

## Expected behavior
Publishing to SNS topic with queue subscribed should result in messages sent to queue

## Actual behavior
When I publish to the SNS topic, the logs show "Removing non-existent queue" and the message does not show up on the SQS queue

# Steps to reproduce
See code:

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

```python
topic = sns.create_topic(Name="test_topic3")
topic_arn = topic["TopicArn"]
test_queue = sqs.create_queue(QueueName="test_queue3")
print(f"Queue url = {test_queue}")
queue_url = test_queue["QueueUrl"]
sns.subscribe(TopicArn=topic_arn, Protocol="sqs", Endpoint=queue_url)
response = sns.publish(TargetArn=topic_arn, Message="Test msg")
response = sqs.receive_message(
        QueueUrl=queue_url,
        AttributeNames=["SentTimestamp"],
        MaxNumberOfMessages=1,
        MessageAttributeNames=["All"],
        VisibilityTimeout=2,
        WaitTimeSeconds=2,
    )
message = response["Messages"][0] # Error - no Messages in response.
```

**Comment by usmangani1** [comment]
@sineqa Thanks for reporting the issue.
We tried to resolve the issue using the same script given above and we are not able to reproduce the issue.We have added the test case https://github.com/localstack/localstack/pull/2886/files here .

Can you please help us with a bit more details on how you are starting the localstack and whole client code and boto3 client configuration.

This will help us to reproduce and resolve the issue sooner.Thanks

**Comment by whummer** [comment]
Thanks for reporting @sineqa . In particular, can you please also share the region configuration on your system (environment variables like `DEFAULT_REGION`, `AWS_DEFAULT_REGION`, etc). Also, can you please try printing the resource ARNs in your code above, and share them with use here? Thanks

**Comment by whummer** [comment]
Any updates on this issue @sineqa ? A small integration test has been added in #2886 - so it is quite likely that the root cause of the error you're seeing could be somehow related to your configuration, most likely a region mismatch. Please provide some details here, we'll be happy to dig into it. Thanks

**Comment by sineqa** [comment]
Thanks - I have been following this with interest. I'm very happy to see the test working which means the issue is likely on my end. I'll do some investigating and report back here when I figure it out.

**Comment by sineqa** [comment]
The region I am using is `ap-southeast-2`. You may be right about the region mismatch - I have suspected that this might be the case. The test code was definitely supplying the region, like this:

```
REGION_NAME = "ap-southeast-2"

sqs = session.client(
        "sqs",
        aws_access_key_id=AWS_ACCESS_KEY,
        aws_secret_access_key=AWS_SECRET_KEY,
        region_name=REGION_NAME,
    )

    sns = session.client(
        "sns",
        aws_access_key_id=AWS_ACCESS_KEY,
        aws_secret_access_key=AWS_SECRET_KEY,
        region_name=REGION_NAME,
        endpoint_url=ENDPOINT_URL,
    )
```

but perhaps localstack or something else in the chain is using a different DEFAULT_REGION or AWS_DEFAULT_REGION.

Meanwhile I have tried renaming my `~/.aws` directory to avoid looking up those credentials when running tests. Then I did `make infra` and then run `AWS_ACCESS_KEY_ID=test AWS_SECRET_ACCESS_KEY=test pytest tests/integration/test_sns.py` but it fails.

```
FAILED tests/integration/test_sns.py::SNSTest::test_publish_sms_endpoint - AssertionError: 3 != 0
FAILED tests/integration/test_sns.py::SNSTest::test_subscribe_platform_endpoint - KeyError: 'arn:aws:sns:us-east-1:000000000000:endpoint/p1/app1/7ee7508e-c13e-4da6-b858-3787a73ff945'
FAILED tests/integration/test_sns.py::SNSTest::test_topic_subscription - KeyError: 'arn:aws:sns:us-east-1:000000000000:TestTopic_snsTest:a854438d-1fbb-489a-a4f5-24c404063b76'
```

**Comment by sineqa** [comment]
Ok so it looks like something odd is going on with regions:
Here is the code:
```
import localstack_client.session

AWS_ACCESS_KEY = "test"
AWS_SECRET_KEY = "test"
REGION_NAME = "ap-southeast-2"
ENDPOINT_URL = "http://localhost:4575"

session = localstack_client.session.Session()
sqs = session.client(
    "sqs",
    aws_access_key_id=AWS_ACCESS_KEY,
    aws_secret_access_key=AWS_SECRET_KEY,
    region_name=REGION_NAME,
)

sns = session.client(
    "sns",
    aws_access_key_id=AWS_ACCESS_KEY,
    aws_secret_access_key=AWS_SECRET_KEY,
    region_name=REGION_NAME,
    endpoint_url=ENDPOINT_URL,
)

topic = sns.create_topic(Name="test_topic3")
topic_arn = topic["TopicArn"]
print(f"Topic ARN = {topic_arn}")
test_queue = sqs.create_queue(QueueName="test_queue3")
queue_url = test_queue["QueueUrl"]
r = sqs.get_queue_attributes(QueueUrl=queue_url, AttributeNames=["QueueArn"])
queue_arn = r["Attributes"]["QueueArn"]
print(f"Queue ARN = {queue_arn}")
```

And here is what it outputs:
```
Topic ARN = arn:aws:sns:ap-southeast-2:000000000000:test_topic3
Queue ARN = arn:aws:sqs:us-east-1:000000000000:test_queue3
```

**Comment by sineqa** [comment]
It seems my `~/.aws` config is getting in the way. If I remove that, I get:
```
Topic ARN = arn:aws:sns:us-east-1:000000000000:test_topic3
Queue ARN = arn:aws:sqs:us-east-1:000000000000:test_queue3
```
regardless of the REGION_NAME that is given to session.client() in the above code (that seems to be ignored no matter what I set it to)

**Comment by sineqa** [comment]
Starting localstack via `AWS_DEFAULT_REGION=ap-southeast-2 DEFAULT_REGION=ap-southeast-2 make infra`

gives me the following:
```
Topic ARN = arn:aws:sns:ap-southeast-2:000000000000:test_topic3
Queue ARN = arn:aws:sqs:us-east-1:000000000000:test_queue3
```

**Comment by whummer** [comment]
Thanks for the update @sineqa . This should now (hopefully) be fixed in #2917 - can you please give it another try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks
